tests/src/test/scala/BinaryTreeTest.scala

```diff
@@ -0,0 +1,235 @@
+/**
+ * Created by Nicolas A Perez (@anicolaspp) on 1/29/16.
+ */
+
+package dogs
+package tests
+
+import dogs.Order.Ordering
+import dogs.Predef._
+import org.scalatest._
+import tests.utils._
+
+class BinaryTreeTestAdd extends FlatSpec with Matchers {
+
+   implicit val cmp =  new Order[Int] {
+    override def apply(l: Int, r: Int): Ordering = l.compareTo(r) match {
+      case 0  => Order.EQ
+      case -1 => Order.LT
+      case 1  => Order.GT
+    }
+  }
```

Comment:
also, you are only covering 3 of the billions of possible values which could be returned by compareTo.  The documentation for this method makes no guarantee that the values will be one of these three.